File: python/3838.weighted-word-mapping.py
class Solution:
    def mapWordWeights(self, words: list[str], weights: list[int]) -> str:
        a = ord("a")
        z = ord("z")
        n = 26

        def worder(w: str) -> str:
            """map 'abcd' -> 'r'"""
            idx = sum(weights[ord(ch) - a] for ch in w) % n
            return chr(z - idx)

        return "".join(map(worder, words))

File: rust/3838.weighted-word-mapping.rs
impl Solution {
    pub fn map_word_weights(words: Vec<String>, weights: Vec<i32>) -> String {
        let n = 26;
        let a = b'a' as u32;
        words
            .into_iter()
            .map(|w| {
                let s = w
                    .bytes()
                    .map(|b| weights[(b - b'a') as usize] as u32)
                    .sum::<u32>();
                (b'z' - (s % 26) as u8) as char
            })
            .collect::<String>()
    }
}
